### ISSUE TITLE
Can someone help me get started in vscode to understand where the typescript features are?

### ISSUE THREAD

**Issue by unphased** [pmizio/typescript-tools.nvim#214]
Sorry for cluttering the issues with this, but Discussions is not enabled so I have no choice but to make an issue. 

So I really want to explore what is possible in terms of refactoring automations. Based on everything I have read vscode is the "gold standard" for typescript in this regard. 

I want to understand how to set up vscode so i can get rich refactoring abilities. In order to compare and think about what neovim may be missing and what might make sense to try to contribute to adding next. 

But I can't figure vscode out. the extensions marketplace is a dumpster fire filled with 2000 typescript garbage extensions. Where is the official one? what is it called? Or maybe it comes with vscode already by default? How do I select some code and open a menu for code actions or refactorings? Obviously I'm happier to stay in the neovim world but I need to expand my horizons to make my neovim as powerful as possible.

I came here via https://github.com/typescript-language-server/typescript-language-server/discussions/850 and https://www.reddit.com/r/neovim/comments/1928vaj/comment/kh32cgg/?utm_source=share&utm_medium=web2x&context=3

Thank you.

**Comment by yioneko** [comment]
VSCode has a code action named "move to file", which implements the functionality you want.

It is currently impossible to implement this according to standard LSP, but possible through defining custom protocol extension.

**Comment by unphased** [comment]
OK i got it figured out. 

I read [here](https://stackoverflow.com/a/77065510) that a sufficiently new typescript is needed. I tried to switch the typescript version from the built in 4.3 something to 5.2.2 that was installed via npm earlier in my project dir. 

That didn't work as after a vscode restart it told me that vscode has been broken and corrupted and i got no typescript capabilities in the menus. 

Then i restarted it again and vscode had decided to update itself this time. And now it comes with typescript 5.3.2 built in. And I was able to get the `Move to file` code action!!!

So now i have a reference point to test this plugin against vscode finally, as well as being able to at least have a fallback so I can open up vscode to do some heavy refactors if i need in the meantime.
